File: app/src/main/java/com/itverse/futuris/data/entities/GroupedElements.java
package com.itverse.futuris.data.entities;


import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Index;
import androidx.room.PrimaryKey;

import static androidx.room.ForeignKey.CASCADE;

@Entity(tableName = "groupedElement_table", foreignKeys = {
        @ForeignKey(entity = Element.class, parentColumns = "element_name",
                childColumns = "element_ID", onUpdate = CASCADE, onDelete = CASCADE)},
        indices = {@Index(value = {"groupedElement_name"}, unique = true)})
public class GroupedElements {

    @ColumnInfo()
    public String groupedElement_name;
    public String element_ID;
    @PrimaryKey(autoGenerate = true)
    public int id;


    public GroupedElements(String groupedElement_name, String element_ID, int id) {
        this.groupedElement_name = groupedElement_name;
        this.element_ID = element_ID;
        this.id = id;
    }
}

File: app/src/main/java/com/itverse/futuris/data/entities/GroupedElements.kt
package com.itverse.futuris.data.entities

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.ColumnInfo

@Entity(tableName = "groupedElement_table")
data class GroupedElements(
    var composantId: Long,
    @field:ColumnInfo
    var name: String,
    @field:PrimaryKey(autoGenerate = true)
    var id: Long = 0
)